app/controllers/foreman_deployments/api/v2/stacks_controller.rb

```diff
@@ -20,6 +20,18 @@ def create
           process_response @stack.save
         end
 
+        api :PUT, '/stacks/:id/', N_('Update imported stack')
+        param_group :stack, :as => :update
+        param :id, :identifier, :required => true
+        def update
+          if @stack.configurations.empty?
```

Comment:
I thought of something less global:
``` Ruby
if @stack.configurations && params[:stack][:definition] && !params[:stack][:configurations]
  render :json => { :error => _("Can't update stack that has been configured") }, :status => :unprocessable_entity
end
ForemanDeployments::StackParser.parse(params[:stack][:definition]) if params[:stack][:definition]
process_response @stack.update_attributes(params[:stack])
```
This way you will be able to remove/change the configurations associated with the stack, but won't be able to update the definitions unless the configs are nullified.